packages/client/src/components/measurements/AgentVote.re

```diff
@@ -0,0 +1,76 @@
+module MeasurementVotes = {
+  type direction =
+    | Down
+    | Up;
+
+  module Styles = {
+    open Css;
+    let vote = style([marginLeft(`px(5)), marginRight(`px(5))]);
+  };
+
+  [@react.component]
+  let make = (~initialVoteCount, ~onUpdate) => {
+    let (count, setCount) = React.useState(() => initialVoteCount);
+    let canIncrement = count < 10;
+    let canDecrement = count > (-10);
+
+    let reducer = (d: direction) => {
+      switch (d) {
+      | Down when canDecrement => count - 1
+      | Up when canIncrement => count + 1
+      | _ => count
+      };
+    };
+
+    let modify = (d: direction) =>
+      setCount(_ => {
+        let newCount = reducer(d);
+        onUpdate(newCount);
+        newCount;
+      });
+
+    <>
+      <Antd.Button
+        size=`small onClick={_ => modify(Down)} disabled={!canDecrement}>
+        {"<" |> Utils.ste}
+      </Antd.Button>
+      <span className=Styles.vote> {string_of_int(count) |> Utils.ste} </span>
+      <Antd.Button
+        size=`small onClick={_ => modify(Up)} disabled={!canIncrement}>
+        {">" |> Utils.ste}
+      </Antd.Button>
+    </>;
+  };
+};
+
+// TODO: I think this would be cleaner if this module were a wrapper, similar to the <Mutation> used inside.
+// I wasn't sure how to do this though.
```

Comment:
I guess we could try to use Reason Apollo Hooks. Maybe it would help us to clean it up.

Reply:
Good point. Do you think hooks could work for most of our mutation logic? The wrapping components seem like a messy method.